npm-scripts.js

```diff
@@ -365,3 +410,62 @@ function installMsysMake()
 
 	executeCmd(`${String(res.stdout).trim()} worker\\scripts\\getmake.py`);
 }
+
+function ensureDir(dir)
+{
+	if (!fs.existsSync(dir))
+	{
+		fs.mkdirSync(dir, { recursive: true });
+	}
+}
+
+function createTar(files, dest)
```

Comment:
Shouldn't this function be `async`? Shouldn't the caller awaits for it to complete? Same concern for functions below.

Reply:
Most are, just not explicitly. I went through and explicitly called them out as async or converted them to return promises if they didn't already.